### ISSUE TITLE
Superclass class_traits are not inherited

### ISSUE THREAD

**Issue by leo60228** [ruffle-rs/ruffle#5292]
### Describe the bug

This breaks https://www.homestuck.com/story/2792?fl=1. Relevant decompilation:

```actionscript
   public class Character extends MovieClip
   {
      public static var dirs:Array = ["Front","Left","Back","Right"];
      
      public var facing:String;
      
      public function Character()
      {
         super();
         this.facing = dirs[int(Math.random() * 4)];
      }
   }

   public dynamic class cg extends Character
   {
      public function cg()
      {
         super();
      }
   }
```

As far as I can tell, this issue stems from the fact that `dirs` is in `Character`'s `class_traits`, which never get searched for a subclass.

### Expected behavior

I expected the page to work (or at least break differently), and for the `dirs` property to be resolved.

### Affected platform

Desktop app

### Operating system

Linux

### Browser

Firefox 92

### Additional information

```
Property does not exist: Multiname { ns: [Package(AvmString { source: Owned("") })], name: Some(AvmString { source: Owned("dirs") }), params: [] }
```

**Comment by leo60228** [comment]
I don't actually think this is *correct*, but this seems to fix the issue:
```diff
diff --git a/core/src/avm2/activation.rs b/core/src/avm2/activation.rs
index 4d693d3d..43ddeb73 100644
--- a/core/src/avm2/activation.rs
+++ b/core/src/avm2/activation.rs
@@ -426,6 +426,10 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
             context,
         };
 
+        if let Some(class) = subclass_object {
+            activation.scope_stack.push(Scope::new(class));
+        }
+
         //Statically verify all non-variadic, provided parameters.
         let arguments_list =
             activation.resolve_parameters(method.method_name(), user_arguments, signature)?;
```

**Comment by leo60228** [comment]
After stubbing `flash.ui.ContextMenu` and `flash.net.URLRequest`, and then patching the swf to not wait for the `Sound` to finish downloading (which it won't, since `URLRequest` is stubbed), the flash loads... and then breaks horribly. Oh well.

**Comment by adrian17** [comment]
Pinging @EmperorBale - in case you think this is in any way related to your PR :)

**Comment by EmperorBale** [comment]
> I don't actually think this is _correct_, but this seems to fix the issue (on top of #5249):
> 
> ```diff
> diff --git a/core/src/avm2/activation.rs b/core/src/avm2/activation.rs
> index 4d693d3d..43ddeb73 100644
> --- a/core/src/avm2/activation.rs
> +++ b/core/src/avm2/activation.rs
> @@ -426,6 +426,10 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
>              context,
>          };
>  
> +        if let Some(class) = subclass_object {
> +            activation.scope_stack.push(Scope::new(class));
> +        }
> +
>          //Statically verify all non-variadic, provided parameters.
>          let arguments_list =
>              activation.resolve_parameters(method.method_name(), user_arguments, signature)?;
> ```

This was *almost* correct, instead of pushing the class on the local scope stack, we should chain it on top of the outer scope.

**Comment by leo60228** [comment]
I think this is closed now?
